apps/settings/js/modules/base/task_scheduler.js

```diff
@@ -0,0 +1,133 @@
+/**
+ * TaskScheduler helps manage tasks and ensures they are executed in
+ * sequential order. When a task of a certain type is enqueued, all pending
+ * tasks of the same type in the queue are removed. This avoids redundant
+ * queries and improves user perceived performance.
+ *
+ * @module Task_sheduler
+ */
+define(function() {
+  'use strict';
+
+  var Defer = require('modules/defer');
+  var Module = require('modules/base/module');
+  var Observable = require('modules/mvvm/observable');
+  var TaskSchedulerConstant = require('modules/base/task_scheduler_constant');
```

Comment:
There is only one type defined and it completely belong to this module. I think we can define it here inside the module.